config/config-api/src/main/java/com/thoughtworks/go/config/GoConfigHolder.java

```diff
@@ -16,18 +16,73 @@
 
 package com.thoughtworks.go.config;
 
+import com.thoughtworks.go.config.remote.PartialConfig;
+import com.thoughtworks.go.util.CachedDigestUtils;
+
+import java.util.List;
+import java.util.Objects;
+
 public class GoConfigHolder {
     public final CruiseConfig config;
-    public final CruiseConfig configForEdit;
-    public CruiseConfig mergedConfigForEdit;
+    private final CruiseConfig configForEdit;
+    private CruiseConfig mergedConfigForEdit;
+    private Checksum checksum;
 
     public GoConfigHolder(CruiseConfig config, CruiseConfig configForEdit) {
+        this(config, configForEdit, null, null);
+    }
+
+    public GoConfigHolder(CruiseConfig config, CruiseConfig configForEdit, CruiseConfig mergedConfigForEdit, List<PartialConfig> partials) {
         this.config = config;
         this.configForEdit = configForEdit;
+        setMergedConfigForEdit(mergedConfigForEdit, partials);
     }
 
-    public GoConfigHolder(CruiseConfig config, CruiseConfig configForEdit, CruiseConfig mergedConfigForEdit) {
-        this(config, configForEdit);
+    private String md5(List<PartialConfig> partials) {
+        if (partials == null || partials.isEmpty()) {
+            return null;
+        }
+        return CachedDigestUtils.md5Hex(partials);
```

Comment:
Is there a reason we don't attempt serialization of `PartialConfig` using MagicalXmlWriter?